### PR TITLE
Update nationalities to use ISO code

### PR DESCRIPTION
**Pull request by chubberlisk**
### Context

The vendor API will expose ISO 3166 country codes for nationalities, ~but we built it to save nationalities as `"British"` et al~.

### Changes proposed in this pull request

~This PR updates how we store nationalities, so instead we use the corresponding ISO code of the country.~

Following https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/335#issuecomment-543077489, we decided to continue storing a candidate's nationality as the nationality _not_ as the ISO code, then when we translate the nationality on the API side. Therefore, this PR just adds the list and refactors to allow for the continue use of `NATIONALITIES` with the help of `NATIONALITY_DEMONYMS` which is an array of just the nationalities (so without the ISO codes).

### Guidance to review

- ~Does what we do in `formatted_nationalities` make sense and it is easy to follow?~
- Is `NATIONALITY_DEMONYMS` a silly name?

### Link to Trello card

[172 - Use ISO codes for Nationality](https://trello.com/c/rPWoApkv/172-use-iso-codes-for-nationality)

### CODE REVIEW

**Comment by tijmenb** [comment]
@chubberlisk and I had an chat about this offline. There's a problem with this system for UK nationalities. If the user chooses "Scottish", we're saving it as "GB" in the database. This means we can't translate it back to "Scottish", since our list translates "GB" to "English", "Scottish", "Welsh", etc.

The solution we've come up with is to save "Scottish" to the database, and translate it into an ISO code when we expose it as an API.

Note for later: this whole business rings alarm bells. We seem to be conflating countries with nationalities (shouldn't nationality be "British"?), and it's a red flag that we're inventing a country-nationality list in this team - I can't imagine this not being a solved problem (alas, the [Nationality pattern](https://github.com/alphagov/govuk-design-system-backlog/issues/73) doesn't give us much). I also have concerns about the usability of the resulting solution. At the very least we should have synonyms in the autocomplete that will allow users to search for "Netherlands" to get to the "Dutch" nationality.

File: config/initializers/nationalities.rb [hunk 1]
```diff
@@ -1,228 +1,228 @@
...
+  %w[GR Greek],
+  %w[GL Greenlandic],
+  %w[GD Grenadian],
+  %w[GU Guamanian],
+  %w[GT Guatemalan],
+  %w[GN Guinean],
+  %w[GY Guyanese],
+  %w[HT Haitian],
+  %w[HN Honduran],
+  ['HK', 'Hong Konger'],
```
Comment (duncanjbrown) [inline config/initializers/nationalities.rb]: This is madness! Could we deactivate the cop for this file?

**Comment by tvararu** [comment]
@duncanjbrown it just would be jarring if you fill in the form as “Scottish” and on the check your answers page you see “British,” not what you chose. Also, it’s only a happy coincidence that you’d even get “British” currently because it’s the first in the alphabetically sorted list...

**Comment by duncanjbrown** [comment]
Sure, but Scottish is not an option at the moment

**Pull request merged**